krabs/krabs/parser.hpp

```diff
@@ -179,13 +179,15 @@ namespace krabs {
             property_info propInfo(pBufferIndex_, currentPropInfo, propertyLength);
             cache_property(pName, propInfo);
 
+            // advance the buffer index since we've already processed this property
+            pBufferIndex_ += propertyLength;
+
             // The property was found, return it
             if (name == pName) {
+                // advance the index since we've already processed this property
+                ++i;
```

Comment:
is this line necessary? if we're about to `return propInfo` then `i` will go out of scope so we don't need to advance it?

Reply:
Note that `i` is defined as a reference to  lastPropertyIndex_` at the start of the for loop.

Reply:
You're right, I missed the ampersand. Thanks!